kernel/src/expressions/scalars.rs

```diff
@@ -546,6 +546,12 @@ impl From<&[u8]> for Scalar {
     }
 }
 
+impl From<bytes::Bytes> for Scalar {
+    fn from(b: bytes::Bytes) -> Self {
+        Self::Binary(b.to_vec())
```

Comment:
Just double checking -- [Bytes::to_vec](https://docs.rs/bytes/latest/bytes/struct.Bytes.html#method.to_vec) is a copying operation provided by `impl Deref<Target=[u8]>`, and we already have `From<&[u8]> for Scalar`? 

If the goal was to transfer ownership cheaply, I don't think that's possible with `Bytes` (which is like `Arc` -- "cheaply cloneable and thereby shareable between an unlimited amount of components").

Is the goal to allow e.g. `b.into()` instead of `b.as_ref().into()`?

Reply:
I went for `Bytes` over `Vec<u8>` mainly because I expect to slice and dice the content of the bytes, which does a copy when working with a raw `Vec`, while `Bytes` avoids this.

Also, when using `ToSchema` to represent a `Vec<u8>`, it will create an `DataType::Array<DataType::Byte>`; therefore, using the `Binary` avoids that. More on this in https://github.com/delta-io/delta-kernel-rs/pull/1361#issuecomment-3368140615

That said, I think we can use `b.into()` here 👍 

```suggestion
impl From<bytes::Bytes> for Scalar {
    fn from(b: bytes::Bytes) -> Self {
        Self::Binary(b.into())
```

Reply:
I think I don't understand the question. We need them both, for each of the types, right?

Reply:
Thanks for the context @scovich, I'll go ahead and merge this PR since I don't think we have to implement any other `From` for Binary-related fields.